plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/CDKConstantUtility.java

```diff
@@ -10,29 +10,143 @@
  ******************************************************************************/ 
 package org.jboss.tools.openshift.cdk.server.core.internal;
 
-import org.eclipse.debug.core.ILaunchConfiguration;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.wst.server.core.IServer;
 import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.IExternalLaunchConstants;
 
-// TODO - allow customization of this location
 public class CDKConstantUtility {
 	private static final String VAGRANT_LOCATION_LINUX = "/usr/bin/vagrant";
-
+	
+	// Seems weird?  See https://github.com/mitchellh/vagrant/issues/1652
+	private static final String VAGRANT_LOCATION_WINDOWS = "C:\\HashiCorp\\Vagrant\\bin\\vagrant.exe";
+	
+	// Linux command to find vagrant full path
+	private static final String LINUX_WHICH_VAGRANT = "which vagrant";
+	
+	// Windows command to find vagrant
+	private static final String WINDOWS_WHERE_VAGRANT = "where vagrant";
+	
+	
+	private static String discoveredVagrantLocation = null;
+	private static boolean searchFailed = false;
+	
+	
 	public static String getVagrantLocation() {
-		return VAGRANT_LOCATION_LINUX;
+		return findVagrantLocation();
```

Comment:
Did some perf-testing on windows and linux.  This method is only a little bit slow when it has to go out to which / where,   and it will only perform this once per workspace, and only if vagrant isn't found in the path already.  Everything's already cached, and it won't try a second time, so the perf. hit is very very very low and the ability to work around path errors is tremendous.  

I believe this patch is for the best.